File: followingApp/app/src/main/java/com/example/idolmastercalendar/IdolAdapter.java
package com.example.idolmastercalendar;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class IdolAdapter extends RecyclerView.Adapter<IdolAdapter.IdolViewHolder> {

    ArrayList<IdolData> idolDataArrayList = new ArrayList<>();

    @NonNull
    @Override
    public IdolViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new IdolViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_entertainer, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull IdolViewHolder holder, int position) {
        ImageView imageView = holder.itemView.findViewById(R.id.idolImage);
        imageView.setImageResource(idolDataArrayList.get(position).getIdolImage());

        final TextView textView = holder.itemView.findViewById(R.id.idolName);
        textView.setText(idolDataArrayList.get(position).idolName);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                textView.setTextColor(Color.parseColor("#6CFF72"));
            }
        });
    }

    @Override
    public int getItemCount() {
        return idolDataArrayList.size();
    }

    class IdolViewHolder extends RecyclerView.ViewHolder {

        public IdolViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: WimiProject/WimiProject/app/src/main/java/com/wimi/idolmaster/ui/choose/idol/IdolAdapter.kt
package com.wimi.idolmaster.ui.choose.idol

import androidx.recyclerview.widget.DiffUtil
import com.wimi.idolmaster.R
import com.wimi.idolmaster.databinding.ItemConceptBinding
import com.wimi.idolmaster.databinding.ItemEntertainerBinding
import com.wimi.idolmaster.domain.model.IdolData
import com.wimi.idolmaster.ui.base.BaseRecyclerAdapter
import com.wimi.idolmaster.ui.base.BaseViewHolder

class IdolAdapter : BaseRecyclerAdapter<IdolData>(DiffCallback()) {

    class DiffCallback : DiffUtil.ItemCallback<IdolData>() {
        override fun areItemsTheSame(oldItem: IdolData, newItem: IdolData): Boolean {
            return oldItem.idolName == newItem.idolName
        }

        override fun areContentsTheSame(oldItem: IdolData, newItem: IdolData): Boolean {
            return oldItem == newItem
        }
    }

    override fun getItemViewType(position: Int) = R.layout.item_entertainer

    override fun onBindViewHolder(holder: BaseViewHolder<IdolData>, position: Int) {
        super.onBindViewHolder(holder, position)

        val binding = holder.getBinding() as ItemEntertainerBinding
        val item = getItem(position)

        binding.idolImage.setImageResource(item.idolImage)
    }

    /*  var idolDataArrayList = ArrayList<IdolData>()

      override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): IdolViewHolder {
          return IdolViewHolder(
              LayoutInflater.from(parent.context)
                  .inflate(R.layout.item_entertainer, parent, false)
          )
      }

      override fun onBindViewHolder(holder: IdolViewHolder, position: Int) {
          val imageView = holder.itemView.findViewById<ImageView>(R.id.idolImage)
          imageView.setImageResource(idolDataArrayList[position].idolImage)
          val textView = holder.itemView.findViewById<TextView>(R.id.idolName)
          textView.text = idolDataArrayList[position].idolName

          holder.itemView.setOnClickListener { //textView.setTextColor(Color.parseColor("#6CFF72"));
              for ((index, idolData) in idolDataArrayList.withIndex()) {
                  idolData.isSelected.set(index == position)
              }
          }
      }

      override fun getItemCount(): Int {
          return idolDataArrayList.size
      }

      inner class IdolViewHolder(itemView: View) :
          RecyclerView.ViewHolder(itemView)*/
}